src/skimage/_build_utils/version.py

```diff
@@ -1,49 +1,56 @@
 #!/usr/bin/env python3
-"""Extract version number from __init__.py"""
 
-import os
+"""Determine and print version number.
 
+Used in top level ``meson.build``.
+"""
 
-def git_version(version):
-    """Append last commit date and hash to version, if available"""
-    import subprocess
-    import os.path
+import subprocess
+from pathlib import Path
 
-    git_hash = ''
+
+def version_in_init():
+    """Extract version string from ``skimage/__init__.py``."""
+    skimage_init = Path(__file__).parent / '../__init__.py'
+    assert skimage_init.is_file()
+
+    with skimage_init.open("r") as file:
+        data = file.readlines()
+
+    version_line = next(line for line in data if line.startswith('__version__ ='))
+    version = version_line.strip().split(' = ')[1].replace('"', '').replace("'", '')
+    return version
+
+
+def try_append_git_commit_n_date(version):
+    """Try to append last commit date and hash to version.
+
+    Apppends nothing if run outside a git repository.
+    """
     try:
-        p = subprocess.Popen(
+        result = subprocess.run(
             ['git', 'log', '-1', '--format="%H %aI"'],
-            stdout=subprocess.PIPE,
-            stderr=subprocess.PIPE,
-            cwd=os.path.dirname(__file__),
+            capture_output=True,
+            cwd=Path(__file__).parent,
```

Comment:
This is a change from before; the `.parent` isn't necessary but is also fine.

Reply:
I didn't actually check if `cwd=` would accept the path to a file, but that would seem somewhat surprising to me. After all, CWD stands for current working _directory_. 

But you are right in questioning this change. I'm actually not sure what behavior would be most appropriate. So I'll keep this behavior as it was before this PR.

Reply:
I think your change was correct, then.